### ISSUE TITLE
Add C Support.

### ISSUE THREAD

**Issue by treeform** [treeform/genny#23]
C Support is nearly there as python has to talk though a C like interface.

**Comment by ArMour85** [comment]
Do you mean C++ right ? Because the readme indicates C support.

**Comment by treeform** [comment]
We are about 99% on the C support. We still have like 1% more to do.

**Comment by ArMour85** [comment]
For the C++ support it will be a simple formality ?

**Comment by treeform** [comment]
Well C++ can use C APIs yes, but I would not call them C++ APIs. For C++ APIs I expect name spaces, classes, operator overloading, polymorphism, exception throwing and maybe RAII. C++ should be treated as a different language. More like Nim or Java.

**Comment by ArMour85** [comment]
I totally agree with you. I hope your project will motivate developers to use Nim to create powerful libraries for other languages. And finally enlarge the community.

**Comment by jdf-id-au** [comment]
Thank you again for your incredible contributions.

This is probably a naïve question. A proc returning a string gives a `char*` return type in the generated C header (after `.cstring` is called in `internal.nim`). Whas is the lifetime of this memory? Isn't the parent `string` vulnerable to garbage collection at Nim's will, and doesn't this affect the `cstring`'s storage? Isn't it necessary to export memory management procs to accompany it (somewhat like `system/allocCStringArray`)?

**Comment by treeform** [comment]
We are aware of the problem with returning strings. We will probably implement some sort of buffer type or make ref/unref work with strings. 

For now it happens to work because pixie is single threaded, returning the string still keeps the data around for C to deal with. But yes as soon as you make another call the data can get clobbered. 

We will fix this.

**Comment by treeform** [comment]
I have fixed the last remaining issue with the returning string support and who owns the memory. I think C backend is now well tested. https://github.com/treeform/genny/pull/66
